### PR TITLE
DOI field improvements

### PR DESCRIPTION
**Pull request by stefan-kolb**
## __Major changes of this PR__
- [x] Removed unused BROWSE action
- [x] Remove double click action for URL and DOI fields
- [x] Add open and fetch from buttons for URL and DOI
- [x] Only activate button if field includes valid DOI/URL

## __New Look__

image

### CODE REVIEW

**Comment by oscargus** [comment]
I use doubleclick and think it is a quite neat feature. Especially when only a single field is available.

**Comment by oscargus** [comment]
Regarding browse I believe it can be useful in combination with filespecific fields (which we currently try to avoid, so I'm fine with the removal).

**Comment by stefan-kolb** [comment]
Double-click might be a neat feature, but is neither intuitive from a UI usage perspective nor shown anywhere. So it is just a feature that can be used by insiders which is really not what I think is our intention.

**Comment by oscargus** [comment]
Maybe. What about a tool tip text then? I would expect requests to get it back if we remove it.

**Comment by stefan-kolb** [comment]
Intuitive to me would be 2 options:
- Adding a open button to the right (ugly)
- Highlighting the text so that one can see it is clickable (problems when one wants to only edit it)

**Comment by Siedlerchr** [comment]
I use that doubleclick feature, too. It is natural to expect Jabref to open the link etc then.
I agree with @oscargus, to me a tooltip à la "Double click to open URL/DOI"

**Comment by oscargus** [comment]
Ahh, I realise that I have been talking about the entry table and not the
entry editor (which I now realise the discussion most likely is about).

Still, I think that a tooltip might do the trick. One can also double click
the date field to set the current date and I think the owner field to
update the owner.

**Comment by stefan-kolb** [comment]
>  It is natural to expect Jabref to open the link etc then.

To be honest I've never experienced that double clicking a text input field opens a link or does anything with the text except marking it.

**Comment by Siedlerchr** [comment]
@stefan-kolb  I thought you meant the Entry Table DOI Field

**Comment by stefan-kolb** [comment]
I mean this DOI field:
image

**Comment by tobiasdiez** [comment]
I would propose to add an inline-button, in the textbox, similar to something like the clear symbol here

which appears on mouse-over and opens the url. Not sure if this can be done with Swing or should wait until JavaFX. Until then an ugly `Open` button might be an interim solution.

**Comment by matthiasgeiger** [comment]
@tobiasdiez We have already discussed with the "clear search" button. Unfortunately, this is no built in function of Swing and requires some quick ugly hacks to implement. But I agree, in general that would be the nicest solution.
However, for the moment I would add an "open" button - especially as @stefan-kolb will already add another button for the "fetch from DOI" logic.

**Comment by stefan-kolb** [comment]
Ok, this is ready for your comments!

File: src/main/java/net/sf/jabref/gui/menus/RightClickMenu.java [hunk 1]
```diff
@@ -171,14 +171,6 @@ public RightClickMenu(JabRefFrame frame, BasePanel panel) {
             }
         });
 
-        add(new GeneralAction(Actions.MERGE_DOI, Localization.lang("Get BibTeX data from DOI")) {
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/menus/RightClickMenu.java]: I so prefer it to stay here (as well). This has been one of my most used features recently.

File: src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java [hunk 2]
```diff
@@ -463,7 +462,7 @@ public void rebuildPanels() {
                     fieldExtras.contains(BibtexSingleFieldProperties.DATEPICKER));
         } else if (fieldExtras.contains(BibtexSingleFieldProperties.EXTERNAL)) {
             // Add external viewer listener for "pdf" and "url" fields.
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java]: I'm wondering if the comment is correct for "pdf" nowadays...

File: src/main/java/net/sf/jabref/gui/entryeditor/FieldExtraComponents.java [hunk 3]
```diff
@@ -15,41 +15,40 @@
...
 import net.sf.jabref.logic.journals.JournalAbbreviationRepository;
 import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.logic.net.URLUtil;
+import net.sf.jabref.logic.util.DOI;
 import net.sf.jabref.logic.util.date.EasyDateFormat;
 import net.sf.jabref.model.database.BibDatabaseMode;
 import net.sf.jabref.model.entry.BibEntry;
 import net.sf.jabref.model.entry.MonthUtil;
 
+import javax.swing.*;
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/entryeditor/FieldExtraComponents.java]: This seems to be yet another ordering of imports. Not a problem as such, just interesting.

**Comment by oscargus** [comment]
Looks good to me! I'd really like to keep the right-click menu though.

With this PR the use of the entry editor has slightly changed from editing to actually using the information (although it was hidden there earlier through double-click). Not sure if it is worth thinking about what we really want the entry editor for.

File: src/main/java/net/sf/jabref/gui/menus/RightClickMenu.java (hunk 1 above)
Comment (stefan-kolb) [inline src/main/java/net/sf/jabref/gui/menus/RightClickMenu.java]: Next step would be to add https://github.com/JabRef/jabref/issues/1098, so I prefer not to add it back in this context menu.

Comment (simonharrer) [inline src/main/java/net/sf/jabref/gui/menus/RightClickMenu.java]: Hm, maybe rename it differently using "autocomplete with BibTeX data from DOI"?

**Comment by simonharrer** [comment]
:+1: LGTM

File: src/main/java/net/sf/jabref/gui/menus/RightClickMenu.java (hunk 1 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/menus/RightClickMenu.java]: I'm not really seeing the problem. A button somewhere else won't help the (superior) habit of right-clicking on an entry and selecting in a menu compared to double-clicking to open the editor, change to the right tab and clicking a button. Marking, click on another button is slightly better, but I still do not see what the problem with the right-click menu staying is. "New from DOI" is not going to end up in the right-click menu? 

I thought the purpose of #1098 was to add e new entry (which IMHO would be better splitting web search and "add/fetch") so I do not see marking and clicking that button as a replacement. 

The context here is that one do get slightly different information when searching and when adding information from DOI, so I usually do both at once. Not to mention improving all my old bib-files.

Comment (simonharrer) [inline src/main/java/net/sf/jabref/gui/menus/RightClickMenu.java]: No worries. :) Action in right click menu stayed in there. :trophy: 

The right click menu now only contains actions related to selected entries in the table. Hence, the "new from doi" was moved to the menu.

**Pull request merged**